core/libraries/rest_api/CalculatedModelFields.php

```diff
@@ -67,7 +70,8 @@ protected function generateNewMapping()
         );
         foreach ($models_with_calculated_fields as $model_name) {
             $calculated_fields_classname = $namespace . $model_name;
-            foreach (array_keys(call_user_func(array($calculated_fields_classname, 'schemaForCalculations'))) as $field_name) {
+            $calculator = LoaderFactory::getLoader()->getShared($calculated_fields_classname);
```

Comment:
> I realize this is additional work but my understanding is whenever we refactor things we're supposed to bring it up to date with current loading best-practices we're using. I know its an even further break of back compat, but it seems to me we've established its okay?

I don't think it's all-or-nothing, because at some point it becomes unreasonable to keep refactoring. Having said that, I think your suggestions are good.
So far we haven't actually broken backward compatibility in this branch. However, Brent has previously ok'd changing constructor arguments that can be managed by the loader, seeing how, if you're using the loader, changing those doesn't break anything.

So I'll take a stab at that.

But what do you think of changing the static methods on the calculators to being non-static? That will break backward compatibility, but pretty up the code some. It's not essential so I didn't do it yet.

Reply:
> I don't think it's all-or-nothing, because at some point it becomes unreasonable to keep refactoring.

My comments were based on my observations of our lead developer's direction in various reviews he's done of my code.

> But what do you think of changing the static methods on the calculators to being non-static? That will break backward compatibility, but pretty up the code some. It's not essential so I didn't do it yet.

Ya breaks back compat but I think its highly unlikely anyone is using those directly.  Let's change the static methods on the calculators to be non-static.
IF we wanted to preserve back compat, we could simply have the static methods left in place temporarily and wrapping constructing the calculator using `LoaderFactory` with a doing_it_wrong notice (and calling the new replacement methods).  But I'm ambivalent on whether we actually do that or not.